projects/ngx-sub-form/src/lib/ngx-sub-form.component.ts

```diff
@@ -188,24 +200,36 @@ export abstract class NgxSubFormComponent<ControlInterface, FormInterface = Cont
 
     const transformedValue: FormInterface = this.transformToFormGroup(obj);
 
-    // for now we throw an error if the transformed value isn't an object with all the expect values
-    // there's an issue to track support for an array https://github.com/cloudnc/ngx-sub-form/issues/9
-    if (Array.isArray(transformedValue)) {
-      throw new ArrayNotTransformedBeforeWriteValueError();
-    }
-
     const missingKeys: (keyof FormInterface)[] = this.getMissingKeys(transformedValue);
     if (missingKeys.length > 0) {
       throw new MissingFormControlsError(missingKeys as string[]);
     }
 
+    this.handleFormArrayControls(transformedValue);
+
     this.formGroup.setValue(transformedValue, {
       emitEvent: false,
     });
     this.formGroup.markAsPristine();
     this.formGroup.markAsUntouched();
   }
 
+  private handleFormArrayControls(obj: any) {
+    Object.entries(obj).forEach(([key, value]) => {
+      if (this.formGroup.get(key) instanceof FormArray && Array.isArray(value)) {
+        const formArray: FormArray = this.formGroup.get(key) as FormArray;
+
+        while (formArray.length > value.length) {
```

Comment:
Add a comment explaining why we use 2 `while` instead of just creating it from scratch and looping over it.

Also add a test to make sure that we reuse existing `FormControl`s within the `FormArray`.